PocketKit/Sources/SharedPocketKit/Session/SessionBackupUtility.swift

```diff
@@ -0,0 +1,113 @@
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Combine
+import RNCryptor
+
+/**
+ A utility class that can back up a `SharedPocketKit.Session` to an encrypted legacy Pocket store.
+ This is done by adding / modifying existing key/value pairs in the store.
+ - Note: The utility of this class is that we will want to force-run the legacy migration
+ after transferring orgs, and a Pocket 8-only user won't have a pre-existing (required) store.
+ By backing up a session, we can create this store (or update a pre-existing one) as necessary.
+ */
+public class SessionBackupUtility {
+    /// The decryption key used in conjunction with `store`.
+    private let password: String
+
+    private let userDefaults: UserDefaults
+    private let store: LegacyEncryptedStore
+    private let notificationCenter: NotificationCenter
+
+    private let notificationQueue = DispatchQueue(label: "SessionBackupUtility", qos: .utility)
+    private var subscriptions: Set<AnyCancellable> = []
+
+    public init(userDefaults: UserDefaults, store: LegacyEncryptedStore, notificationCenter: NotificationCenter) {
+        self.userDefaults = userDefaults
+        self.store = store
+        self.notificationCenter = notificationCenter
+
+        // Return an existing decryption key, else create + store a new one, and return that
+        if let existing = userDefaults.string(forKey: LegacyUserMigration.decryptionKey) {
+            self.password = existing
+        } else {
+            // Generate a password the same way as the legacy app
+            let password = RNCryptor.randomData(ofLength: 32).base64EncodedString()
+            userDefaults.set(password, forKey: LegacyUserMigration.decryptionKey)
+            self.password = password
+        }
+    }
+
+    public func start() {
+        // Utilize NotificationCenter to save a user session on-disk after a user has logged in
+        notificationCenter.publisher(for: .userLoggedIn)
+            .receive(on: notificationQueue)
+            .sink { [weak self] notification in
+                guard let session = notification.object as? Session else {
+                    return
+                }
+
+                try? self?.save(session: session)
+            }.store(in: &subscriptions)
+
+        // Utilize NotificationCenter to reset a user session on-disk after a user has logged out
+        notificationCenter.publisher(for: .userLoggedOut)
+            .receive(on: notificationQueue)
+            .sink { [weak self] notification in
+                try? self?.save(session: nil)
+            }.store(in: &subscriptions)
+    }
+
+    /// Saves a `SharedPocketKit.Session` to `store`, adding / modifying existing key/value pairs
+    ///  within the store to contain the requested Session data.
+    private func save(session: Session?) throws {
+        let data: Data
+        do {
+            // `decryptStore` will throw if the on-disk store was
```

Comment:
Do we need to log or send any of these states to sentry?

Reply:
We might want to capture what's in the `catch`, even though it's technically a valid code path. Is a breadcrumb with the error enough, here? I don't want to bloat the issues with errors when it could possibly just be because the file didn't exist on-disk (which is legitimate - a fresh install of Pocket won't have it).